dns: support for custom hosts

FabianPonce opened the pull request:
This PR adds support for passing host resolver information to Lima, allowing the internal DNS resolver to resolve any custom names you wish. This is particularly useful for services that do not use `getaddrinfo`, ie. specific nginx, envoy configurations.

abiosoft commented:
Hi, thanks for this.

Rather than bringing the entire `hostResolver` config into Colima, I would recommend supporting config for only the custom hosts. 

It can easily be an array flag e.g. `--custom-host=host.docker.internal=host.lima.internal`
And a config section for `customHosts` that is a `map[string]string`.

We would also need to keep `host.docker.internal` as default/base value and only append the user specified values.

FabianPonce commented:
I agree, I think that would be a much cleaner solution in general. I'll make some changes and re-request review.

FabianPonce commented:
@abiosoft I felt that the custom hosts functionality, being related to DNS, should live somewhere near the resolver settings, so I moved this under the network section. I'm happy to break it out into the originally suggested location if you still think that's more suitable. 

I also renamed the DNS Resolvers list in the config struct to `DNSResolvers` to make that more clear in the code. I left the naming in the yaml alone as `network.dns`, though, to keep backwards compatibility for people using that.
